### PR TITLE
Farcaster member directory property and required login identity

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/allow-space-admins-to-require-members-to-connect-farcaster-identity-4995024008006026)

### CODE REVIEW

File: components/login/components/WarpcastLogin.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import type { LoggedInUser } from 'models';
 
 import { FarcasterLoginModal } from './FarcasterModal';
 
-function WarpcastLoginButton({ type }: { type: LoginType }) {
+export function useWarpcastLogin({ type }: { type: LoginType }) {
```
Comment (mattcasey) [inline components/login/components/WarpcastLogin.tsx:15]: if this is a hook, it should be in a hook file

**Review by mattcasey — CHANGES_REQUESTED** [review]
this adds .02MB to the payload, looks like it is probably adding the warpcast SDK to our app. please use lazy loading like we are doing with LoginButton

File: components/_app/components/ConnectedAccounts.tsx [hunk 2]
```diff
@@ -1,10 +1,13 @@
+import { AuthKitProvider } from '@farcaster/auth-kit';
 import { Stack, Typography } from '@mui/material';
 import { type ReactNode } from 'react';
 
 import { useAddUserWallets } from 'charmClient/hooks/profile';
 import { Button } from 'components/common/Button';
 import { FieldWrapper } from 'components/common/form/fields/FieldWrapper';
+import { FarcasterLoginModal } from 'components/login/components/FarcasterModal';
 import { WalletSign } from 'components/login/components/WalletSign';
+import { useWarpcastLogin } from 'components/login/components/WarpcastLogin';
```
Comment (mattcasey) [inline components/_app/components/ConnectedAccounts.tsx:10]: this should be lazy-loaded

File: components/members/components/MemberDirectorySidebar/MemberPropertyItem.tsx [hunk 3]
```diff
@@ -14,6 +15,29 @@ import { iconForPropertyType } from 'components/common/DatabaseEditor/widgets/ic
...
 import { MEMBER_PROPERTY_CONFIG, PREMIUM_MEMBER_PROPERTIES } from 'lib/members/constants';
 import DiscordIcon from 'public/images/logos/discord_logo.svg';
+import FarcasterLogo from 'public/images/logos/farcaster.svg';
+
+function FarcasterIcon() {
+  const theme = useTheme();
+
+  if (theme.palette.mode === 'dark') {
+    return (
+      <SvgIcon viewBox='0 0 20 20' fontSize='small'>
```
Comment (mattcasey) [inline components/members/components/MemberDirectorySidebar/MemberPropertyItem.tsx:25]: we should be using CSS to modify the SVG based on light mode. using JS is very hacky, but not a blocker

File: lib/members/getSpaceMembers.ts [hunk 4]
```diff
@@ -142,7 +143,8 @@ export async function getSpaceMembers({
           properties,
           searchValue: getMemberSearchValue(spaceRole.user, visiblePropertiesMap, memberUsernameRecord[spaceRole.id]),
           roles,
-          isBot: userData.isBot ?? undefined
+          isBot: userData.isBot ?? undefined,
+          farcasterUser: userData.farcasterUser || undefined
```
Comment (mattcasey) [inline lib/members/getSpaceMembers.ts:147]: please map this to the fields you need. its a bad practice to just send the whole JSON forward, also requires type-casting

File: lib/members/interfaces.ts [hunk 5]
```diff
@@ -54,6 +55,7 @@ export type Member = Pick<User, 'id' | 'createdAt' | 'updatedAt' | 'username'> &
   roles: Pick<Role, 'name' | 'id'>[];
   onboarded: boolean;
   searchValue: string;
+  farcasterUser?: FarcasterUser;
```
Comment (mattcasey) [inline lib/members/interfaces.ts:58]: just pick the fields we need here

**Review by mattcasey — CHANGES_REQUESTED** [review]
Required changes requested:
- dont send the entire farcaster blob to the frontend
- lazy-load the auth kit

File: components/login/components/WarpcastLogin.tsx (hunk 1 above)
Comment (Devorein) [reply components/login/components/WarpcastLogin.tsx:15]: Updated

File: components/_app/components/ConnectedAccounts.tsx (hunk 2 above)
Comment (Devorein) [reply components/_app/components/ConnectedAccounts.tsx:10]: Updated

File: components/members/components/MemberDirectorySidebar/MemberPropertyItem.tsx (hunk 3 above)
Comment (Devorein) [reply components/members/components/MemberDirectorySidebar/MemberPropertyItem.tsx:25]: Removed the explicit fill value and it worked for both theme.

File: lib/members/interfaces.ts (hunk 5 above)
Comment (Devorein) [reply lib/members/interfaces.ts:58]: Updated

**Pull request merged**